CMakeLists.txt

```diff
@@ -806,6 +806,35 @@ add_custom_target(check
   COMMAND ${CMAKE_CTEST_COMMAND}
   DEPENDS tests)
 
+# Catch2 C++ Unit Test support:
+option(WITH_CATCH2 "Enable Catch2 unit tests" ON)
+option(PREFER_SYSTEM_CATCH2 "Enable Catch2 unit tests (use system packages, if found)" OFF)
```

Comment:
why is this not the default? does it break if v2 packages are installed, or fall back to building 3.8.1 from source?

Reply:
if the presence of v2 packages breaks this, we probably shouldn't even expose this option and rely only WITH_SYSTEM_CATCH2 (again, for consistency with other dependencies)

Reply:
We can make that the default, if desirable. Looking at the trace, what's normal for CPM is to first try to get the exact version you ask for from the source you request (in this case, GitHub); local packages are a fallback. That makes sense to me for most projects. The switches, then, are changing that by saying to try the system packages first (PREFER) and to only use the system packages (i.e. do not download), "REQUIRE".

Reply:
@cbodley Ok-- please have a look at the latest organization. If you are still thinking we should drop PREFER_SYSTEM_CATCH2, I can certainly do that, though I don't see /harm/ leaving it in. But maybe a black-and-white option is indeed going to cause less confusion (I can see someone thinking they linked/used one when it was actually the other). Thank you!

Reply:
Dropped!